File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(string t, string p) {
    int answer = 0;
    
    int t_size = t.size();
    int p_size = p.size();
    
    long long p_num = stoll(p);
    
    for(int i = 0 ; i < t_size - p_size + 1 ; i++){
        string cur = t.substr(i, p_size);
        long long cur_num = stoll(cur);
        if(cur_num <= p_num)
            answer++;
    }
    
    return answer;
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int pLength = p.length();
        
        for (int i = 0; i <= t.length() - pLength; i++) {
            String subStr = t.substring(i, i + pLength);
            if (subStr.compareTo(p) <= 0) {
                answer++;
            }
        }
        
        return answer;
    }
}
